definitions/features/service.rb

```diff
@@ -61,29 +64,27 @@ def use_system_service(action, options, spinner)
   def run_action_on_services(action, options, spinner)
     status = 0
     failed_services = []
-
-    filtered_services(options).each do |service|
-      spinner.update("#{action_noun(action)} #{service}")
-      item_status, output = service.send(action.to_sym)
-
-      formatted = format_status(output, item_status, options)
-      puts formatted unless formatted.empty?
-
-      if item_status > 0
-        status = item_status
-        failed_services << service
+    filtered_services(options).each_value do |group|
+      group.each_slice(2).to_a.each do |subgroup|
+        spinner.update("#{action_noun(action)} #{subgroup.join(', ')}")
+      end
+      result = execute_with_status('systemctl ' \
+        "#{action} #{group.map(&:name).join(' ')}")
+      if result[0] > 0 || action == 'status'
```

Comment:
One query - why did you add special handling for action == `status` here and inside `format_status` method? As per my understanding it is because you clubbed two things - failed_services & formatting the message logic in a single method, right?

I found it little confusing while reading the code, will it be possible to separate out these two things

Reply:
Yes, I agree its bit confusing. Actually as services are being handled in group I need to sequentially get the status of every service if there is failure or action is status. I dont see a way to simply it more than this, at max I can remove the OR condition on line 73 and use elsif because it exactly needs same methods with different params.
With this I better I change failed_services variable to `statuses` as it resembles this implementation.